diesel/src/query_builder/select_by_statement/mod.rs

```diff
@@ -0,0 +1,100 @@
+use super::{AstPass, QueryFragment};
+use crate::backend::Backend;
+use crate::connection::Connection;
+use crate::deserialize::{QueryableByName, TableQueryable};
+// use crate::expression::subselect::ValidSubselect;
+use crate::expression::*;
+use crate::query_builder::{QueryId, SelectQuery};
+use crate::query_dsl::LoadQuery;
+// use crate::query_source::joins::{AppendSelection, Inner, Join};
+use crate::query_source::*;
+use crate::result::QueryResult;
+
+mod dsl_impls;
+
+/// it is like `SelectStatement` but without
+/// `Query`, `ValidSubselect` and `AppendSelection`
+#[derive(Debug, Clone, Copy)]
+#[doc(hidden)]
+#[must_use = "Queries are only executed when calling `load`, `get_result` or similar."]
+pub struct SelectByStatement<Selection, Statement> {
+    pub(crate) select: std::marker::PhantomData<Selection>,
+    pub(crate) inner: Statement,
+}
+
+impl<S, STMT> QueryId for SelectByStatement<S, STMT>
+where
+    STMT: QueryId,
+{
+    type QueryId = SelectByStatement<(), STMT::QueryId>;
+    const HAS_STATIC_QUERY_ID: bool = STMT::HAS_STATIC_QUERY_ID;
+}
+
+impl<S, ST> SelectByStatement<S, ST> {
+    pub(crate) fn new(inner: ST) -> Self {
+        SelectByStatement {
+            inner,
+            select: Default::default(),
+        }
+    }
+}
+
+impl<DB, S, STMT> QueryFragment<DB> for SelectByStatement<S, STMT>
+where
+    DB: Backend,
+    STMT: QueryFragment<DB>,
+{
+    fn walk_ast(&self, mut out: AstPass<DB>) -> QueryResult<()> {
+        self.inner.walk_ast(out.reborrow())?;
+        Ok(())
+    }
+}
+
+impl<ST, S, STMT> SelectQuery for SelectByStatement<S, STMT>
+where
+    S: TableQueryable,
+    S::Columns: Expression<SqlType = ST>,
+    STMT: SelectQuery<SqlType = ST>,
+{
+    // TODO: check SelectByClause
+    type SqlType = ST;
+}
```

Comment:
Just noting: That's the super explicit variant. Just forwarding it to `STMT::SqlType` would also be fine.